packages/admin-panel/src/VizBuilderApp/components/DataLibrary/component/TransformSelectedOptionWithJsonEditor.js

```diff
@@ -0,0 +1,138 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Checkbox } from '@tupaia/ui-components/src/components/DataLibrary/Checkbox';
+import DownArrow from '@material-ui/icons/ArrowDropDown';
+import CheckBoxOutlinedIcon from '@material-ui/icons/CheckBoxOutlined';
+import styled from 'styled-components';
+import { BaseSelectedOption, FlexSpaceBetween, JsonEditor } from '@tupaia/ui-components/';
+
+const FlexBetweenPanel = styled(FlexSpaceBetween)`
+  width: 100%;
+`;
+
+const JsonEditorPanel = styled.div`
+  flex: 1;
+  display: flex;
+  width: 100%;
+
+  > div {
+    width: 100%;
+  }
+
+  .jsoneditor {
+    border: none;
+    cursor: text;
+  }
+`;
+
+const OptionPanelWithJsonEditor = styled.div`
+  display: flex;
+  justify-content: space-between;
+  flex-direction: column;
+  align-items: flex-start;
+  height: auto;
+  flex: 1;
+  min-width: 0;
+  overflow: hidden;
+`;
+
+const DownArrowIconWrapper = styled.div`
+  display: flex;
+
+  .icon-wrapper {
+    cursor: pointer;
+  }
+  .MuiSvgIcon-root {
+    transition: transform 0.3s ease;
+    transform: rotate(${({ $expanded }) => ($expanded ? '0deg' : '-90deg')});
+  }
+`;
+
+export const TransformSelectedOptionWithJsonEditor = ({
```

Comment:
In fact, most of this component is almost identical to the `AggregateSelectedOptionWithJsonEditor`. Could we look at a way that we could refactor them into common code?

Reply:
Have refactored it to reuse the code. Nice call!